subsys/modem/backends/Kconfig

```diff
@@ -21,4 +21,12 @@ config MODEM_BACKEND_UART_ASYNC
 	bool "Modem UART backend module async implementation"
 	default y if UART_ASYNC_API
 
+if MODEM_BACKEND_UART_ASYNC
+
+config MODEM_BACKEND_UART_ASYNC_TRANSMIT_TIMEOUT_US
+	int "Modem UART async transmit timeout in microseconds"
+	default 100000
```

Comment:
How about milliseconds? Doubt you need microsecond precision for this.

Reply:
Agreed :)